### PR TITLE
✨ Add Bottlerocket kernel settings

### PR DESCRIPTION
**Pull request by ahreehong**
**What this PR does / why we need it**:
Adds support for configuring kernel settings on Bottlerocket

```yaml
   bottlerocket:
      kernel:
        sysctlSettings:
          foo: bar
          abc: def
```
This gets parsed to bottlerocket as `settings.kernel.sysctl`

### CODE REVIEW

File: bootstrap/kubeadm/api/v1beta1/kubeadm_types.go [hunk 1]
```diff
@@ -197,6 +197,10 @@ type ClusterConfiguration struct {
 type BottlerocketSettings struct {
 	// Kubernetes holds the kubernetes settings for bottlerocket nodes.
 	Kubernetes *BottlerocketKubernetesSettings `json:"kubernetes,omitempty"`
+
+	// KernelSettings contains additional kernel settings for Bottlerocket.
+	// +optional
+	KernelSettings map[string]string `json:"kernelSettings,omitempty"`
```
Comment (vignesh-goutham) [inline bootstrap/kubeadm/api/v1beta1/kubeadm_types.go]: Kernel settings might involve a lot more than we are exposing here, so how about we make it a struct and expose sysctls under it
```
type BottlerocketKernelSettings struct {
    Sysctl map[string]string
}
```

File: bootstrap/kubeadm/api/v1beta1/kubeadm_types.go [hunk 2]
```diff
@@ -212,6 +216,12 @@ type BottlerocketKubernetesSettings struct {
 	ClusterDNSIPs []string `json:"clusterDNSIPs,omitempty"`
 }
 
+// BottlerocketKernelSettings holds the kernel settings for bottlerocket nodes
+type BottlerocketKernelSettings struct {
+	// SysctlSettings defines the kernel sysctl settings to set for bottlerocket nodes.
+	SysctlSettings map[string]string `json:"sysctlSettings,omitempty"`
```
Comment (g-gaston) [inline bootstrap/kubeadm/api/v1beta1/kubeadm_types.go:222]: Why can't this be structured like the kubernetes settings?

Comment (ahreehong) [reply bootstrap/kubeadm/api/v1beta1/kubeadm_types.go:222]: The kubernetes settings set the list it takes in as the value for the `allowed-unsafe-sysctls` key. The kernel settings takes in key/value pairs and gets set like this instead:
```
[settings.kernel.sysctl]
"key" = "value"
```
So we need to structure it in a way to pass in both key and value

File: bootstrap/kubeadm/internal/bottlerocket/bottlerocket.go [hunk 3]
```diff
@@ -296,6 +306,17 @@ func parseNodeLabels(nodeLabels string) string {
 	return nodeLabelsToml
 }
 
+func parseSysctlSettings(sysctlSettings map[string]string) string {
+	if sysctlSettings == nil {
+		return ""
+	}
```
Comment (abhinavmpandey08) [inline bootstrap/kubeadm/internal/bottlerocket/bottlerocket.go]: nit: The for loop below should handle the case when `sysctlSettings` is nil. There just won't be any loop iterations when sysctlSettings is nil. So we shouldn't need an explicit check

**Review by abhinavmpandey08 — APPROVED** [review]

**Review by abhinavmpandey08 — APPROVED** [review]

**Pull request merged**